File: app/src/main/java/com/example/pearvideoclient/channel/adapter/RankContAdapter.java
package com.example.pearvideoclient.channel.adapter;

import android.app.Activity;
import android.content.Intent;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.core.app.ActivityOptionsCompat;
import androidx.core.util.Pair;
import androidx.core.view.ViewCompat;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.pearvideoclient.R;
import com.example.pearvideoclient.content.ContentActivity;
import com.example.pearvideoclient.entity.CategoryContsBean;
import com.example.pearvideoclient.utils.GlideUtils;

import java.util.List;

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-30 11:29
 * @ClassName: RankContAdapter
 */
public class RankContAdapter extends BaseQuickAdapter<CategoryContsBean.ContListBean, BaseViewHolder> {
    public RankContAdapter(int layoutResId, @Nullable List<CategoryContsBean.ContListBean> data) {
        super(layoutResId, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, CategoryContsBean.ContListBean item) {
        helper.setText(R.id.tv_video_name, item.getName())
                .setText(R.id.tv_video_author_duration, item.getUserInfo().getNickname() + " | " + item.getDuration())
                .setText(R.id.tv_rank, (helper.getAdapterPosition() + 1) + "");

        if (helper.getAdapterPosition() > 2) {
            helper.setVisible(R.id.tv_rank, false);
        } else {
            helper.setVisible(R.id.tv_rank, true);
        }
        ImageView ivVideoImg = helper.getView(R.id.iv_video_img);
        GlideUtils.loadWithPlaceHolder(item.getPic(), ivVideoImg, null, null);

        TextView textView = helper.getView(R.id.tv_video_name);
        textView.setTransitionName("textView");
        helper.getView(R.id.rl_parent).setOnClickListener(v -> {
            Intent intent = new Intent(mContext, ContentActivity.class);
            intent.putExtra("contId", item.getContId());
            intent.putExtra("userId", item.getUserInfo().getUserId());
            Pair<View, String> namePair = new Pair<>(textView, ViewCompat.getTransitionName(textView));

            ActivityOptionsCompat option = ActivityOptionsCompat.makeSceneTransitionAnimation(
                    (Activity) mContext, namePair);
            mContext.startActivity(intent, option.toBundle());
        });
    }
}

File: app/src/main/java/com/example/pearvideoclient/channel/adapter/RankContAdapter.kt
package com.example.pearvideoclient.channel.adapter

import android.app.Activity
import android.content.Intent
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.core.app.ActivityOptionsCompat
import androidx.core.util.Pair
import androidx.core.view.ViewCompat

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.example.pearvideoclient.R
import com.example.pearvideoclient.content.ContentActivity
import com.example.pearvideoclient.entity.CategoryContsBean
import com.example.pearvideoclient.utils.GlideUtils

/**
 * @Description: java类作用描述
 * @Author: jiazhu
 * @Date: 2019-08-30 11:29
 * @ClassName: RankContAdapter
 */
class RankContAdapter(layoutResId: Int, data: List<CategoryContsBean.ContListBean>?) :
        BaseQuickAdapter<CategoryContsBean.ContListBean, BaseViewHolder>(layoutResId, data) {

    override fun convert(helper: BaseViewHolder, item: CategoryContsBean.ContListBean) {
        helper.setText(R.id.tv_video_name, item.name)
                .setText(R.id.tv_video_author_duration, item.userInfo.nickname + " | " + item.duration)
                .setText(R.id.tv_rank, (helper.adapterPosition + 1).toString() + "")

        if (helper.adapterPosition > 2) {
            helper.setVisible(R.id.tv_rank, false)
        } else {
            helper.setVisible(R.id.tv_rank, true)
        }
        val ivVideoImg = helper.getView<ImageView>(R.id.iv_video_img)
        GlideUtils.loadWithPlaceHolder(item.pic, ivVideoImg, null, null)

        val textView = helper.getView<TextView>(R.id.tv_video_name)
        textView.transitionName = "textView"
        helper.getView<View>(R.id.rl_parent).setOnClickListener {
            val intent = Intent(mContext, ContentActivity::class.java)
            intent.putExtra("contId", item.contId)
            intent.putExtra("userId", item.userInfo.userId)
            val namePair = Pair<View, String>(textView, ViewCompat.getTransitionName(textView))

            val option = ActivityOptionsCompat.makeSceneTransitionAnimation(
                    mContext as Activity, namePair)
            mContext.startActivity(intent, option.toBundle())
        }
    }
}
